pkg/resources/resource_user.go

```diff
@@ -123,7 +123,8 @@ func userRead(ctx context.Context, d *schema.ResourceData, meta interface{}) dia
 
 	userResponse, err := frontegg.ReadUser(ctx, client, userID)
 	if err != nil {
-		if strings.Contains(err.Error(), "not found") {
+		if strings.Contains(err.Error(), "404") || strings.Contains(strings.ToLower(err.Error()), "not found") {
+			// User doesn't exist, remove from state
```

Comment:
Digging into the Frontegg helpers, it seems as if we juggling this a few ways:

1. `ReadUser` calls `doRequest`, [which returns](https://github.com/MaterializeInc/terraform-provider-materialize/blob/main/pkg/frontegg/helpers.go#L31-L39) `nil, fmt.Errorf(...)` for 404 errors.
2. Since `err != nil`, [we immediately return](https://github.com/MaterializeInc/terraform-provider-materialize/blob/08b526afd8a7c0c44e07953ecd6d6d071220ec34/pkg/frontegg/user.go#L90-L92) that same result from `ReadUser`
3. Further along in `ReadUser` (after we've early-returned out), [we invoke an API error handler](https://github.com/MaterializeInc/terraform-provider-materialize/blob/08b526afd8a7c0c44e07953ecd6d6d071220ec34/pkg/frontegg/user.go#L95-L97) for non-200 status codes and return a string with different formatting from the Frontegg helper in step 1.

Ideally we can consolidate this string formatting to be consistent across endpoints, or even include a status code (by using an Error struct).

Reply:
Good catch @arusahni! Thank you for this!

I've created a tracking issue for this: https://github.com/MaterializeInc/terraform-provider-materialize/issues/641